crates/generator/src/account_lock_manage/secp256k1.rs

```diff
@@ -58,3 +59,71 @@ impl LockAlgorithm for Secp256k1 {
         Ok(true)
     }
 }
+
+#[derive(Debug, Default)]
+pub struct Secp256k1Eth;
+
+/// Usage
+/// register AlwaysSuccess to AccountLockManage
+///
+/// manage.register_lock_algorithm(code_hash, Box::new(AlwaysSuccess::default()));
+impl LockAlgorithm for Secp256k1Eth {
+    fn verify_signature(
+        &self,
+        lock_args: Bytes,
+        signature: Signature,
+        message: H256,
+    ) -> Result<bool, LockAlgorithmError> {
+        if lock_args.len() != 20 {
+            return Err(LockAlgorithmError::InvalidLockArgs);
+        }
+        let mut expected_pubkey_hash = [0u8; 20];
+        expected_pubkey_hash.copy_from_slice(&lock_args);
+        let signature: RecoverableSignature = {
+            let signature: [u8; 65] = signature.unpack();
+            let recid = RecoveryId::from_i32(signature[64] as i32)
+                .map_err(|_| LockAlgorithmError::InvalidSignature)?;
+            let data = &signature[..64];
+            RecoverableSignature::from_compact(data, recid)
+                .map_err(|_| LockAlgorithmError::InvalidSignature)?
+        };
+        let msg = secp256k1::Message::from_slice(message.as_slice())
+            .map_err(|_| LockAlgorithmError::InvalidSignature)?;
+        let pubkey = SECP256K1
+            .recover(&msg, &signature)
+            .map_err(|_| LockAlgorithmError::InvalidSignature)?;
+        let pubkey_hash = {
+            let mut hasher = Keccak256::new();
+            hasher.update(&pubkey.serialize_uncompressed()[1..]);
+            let buf = hasher.finalize();
+            let mut pubkey_hash = [0u8; 20];
+            pubkey_hash.copy_from_slice(&buf[12..]);
+            pubkey_hash
+        };
+        if &pubkey_hash != &expected_pubkey_hash {
+            return Ok(false);
+        }
+        Ok(true)
+    }
+}
+
+#[test]
+fn test_secp256k1_eth() {
+    let mut hasher = Keccak256::new();
+    hasher.update("\x19Ethereum Signed Message:\n32");
```

Comment:
Because Rollup passes a unified format message to the different signature algorithms, we should hash the prefix("\x19Ethereum Signed Message:\n | message.len()") in the Secp256k1Eth::verify_signature method.

Reply:
Agreed, this is now fixed.